Composer/packages/client/src/store/index.tsx

```diff
@@ -28,6 +30,32 @@ import filePersistence from './persistence/FilePersistence';
 
 const { defaultFileResolver } = TemplatesParser;
 
+const getUserSettings = (): UserSettings => {
+  const defaultSettings = {
+    appUpdater: {
+      autoDownload: false,
+      useNightly: true,
+    },
+    codeEditor: {
+      lineNumbers: false,
+      wordWrap: false,
+      minimap: false,
+    },
+    propertyEditorWidth: 400,
+    dialogNavWidth: 180,
+  };
+  const loadedSettings = storage.get('userSettings') || {};
+  const settings = merge(defaultSettings, loadedSettings);
+
+  if (isElectron()) {
+    // push the settings to the electron main process
+    const { ipcRenderer } = window as any;
```

Comment:
I think we can go ahead and add ipcRenderer to the window namespace. Let's create a file in the `types` directory called `window.d.ts`:

```ts
declare global {
  interface IPCRenderer {
    send: (type: string, payload: any) => void;
  }

  interface Window {
    ipcRenderer: IPCRenderer;
    __IS_ELECTRON__: boolean;
  }
}
```